File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint/tp_dynamic_eval_construct.py
"""TP: WS5 dynamic-construct surfacing. `eval(x)` reached with a tainted,
non-constant operand is undecidable for static taint -- must surface an
explicit needs-review dynamic_construct finding (CWE-470)."""


def run():
    code = request.args.get("code")
    eval(code)

File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint_js/tp_dynamic_eval_construct.js
// TP: WS5 dynamic-construct surfacing. `eval(x)` reached with a tainted,
// non-constant operand is undecidable for static taint (could run
// anything) -- must surface an explicit needs-review dynamic_construct
// finding (CWE-470), independent of whatever the normal sink pass does.
app.get('/run', (req, res) => {
    const code = req.query.code;
    eval(code);
    res.send('done');
});
